src/generic/datavgen.cpp

```diff
@@ -2149,6 +2298,9 @@ void wxDataViewMainWindow::OnLeave()
 
 wxBitmap wxDataViewMainWindow::CreateItemBitmap( unsigned int row, int &indent )
 {
+    if (row < 0 || row >= GetRowCount())
+        return wxNullBitmap;
+
```

Comment:
This seems like an unrelated change, it would be better to extract it into a separate commit.

I also wonder if we really should silently ignore calls to this function with invalid arguments, i.e. if this shouldn't be a `wxCHECK()` instead. Could you please explain when exactly does this happen?

Reply:
It should be a separate commit.

Maybe wxCHECK here will be better here, but I am not sure.
I think, wxCHECK + the code above should be the best.

There is a segfault when you try to drag the root node of the tree. Actually when you trying to drag the background of the window.

Reply:
If this code can be triggered by a user action, it shouldn't be a `wxCHECK` -- those are for programmer's errors and nothing the user does should be able to trigger them.

Looking at the code, I think the real bug is that we start dragging from invalid row in the first place. I.e. `wxDataViewDropSource` shouldn't be created at all in `wxDataViewMainWindow::OnMouse()` if the row is invalid because why could dragging "nothing" be ever useful anyhow?